src/environment.ts

```diff
@@ -58,15 +64,27 @@ export async function prepareEnvironment() {
   // Ensure Python (>= 3.8) and pip are installed
   await checkPythonPrerequisites(3, 8);
 
-  // Install latest version of commisery
-  await exec.exec("python3", [
-    "-m",
-    "pip",
-    "install",
-    "--upgrade",
-    "--requirement",
-    path.join(__dirname, "requirements.txt"),
-  ]);
+  const requirements_file = path.join(__dirname, "requirements.txt");
+  const cache_hash = await glob.hashFiles(requirements_file);
+  const cache_key = `python-3-8-${os.arch()}-venv-commisery-${cache_hash}`;
```

Comment:
I can imagine this breaking when a cache for a different Python version is re-used.
Maybe `exec('python3 --version').stdout`?